lib/jsonapi/utils/string.ex

```diff
@@ -109,12 +112,19 @@ defmodule JSONAPI.Utils.String do
     with words <-
            Regex.split(
              ~r{(?<=[a-zA-Z0-9])[-_](?=[a-zA-Z0-9])},
-             to_string(value)
+             to_string(value),
+             trim: true
```

Comment:
Annotation: Using `trim: true` means we can remove the need to manually filter out the empty strings.